### ISSUE TITLE
json_encode() never returns FALSE

### ISSUE THREAD

**Issue by matbech** [peachpiecompiler/peachpie#892]
With the following code, the compiler emits a warning:
warning PHP5011: Unreachable code detected
However, I believe this is a false positive. json_encode can return false according to the documentation:
https://www.php.net/manual/en/function.json-encode.php

```
 $requestBody = json_encode($input);
        if ($requestBody === false) {
-->            throw new InvalidInputException(
                'Error encoding input as JSON: '
                . $this->jsonErrorDescription()
            );
        }
```

The same issue for the tempnam function:
```
        $tmp = @tempnam($dir, 'wsw');

        if ($tmp === false) {
-->            throw Exception\RuntimeException::unableToCreateTemporaryFile($dir);
        }
```

**Comment by jakubmisek** [comment]
Good point! the actual warning is correct since the implementation of `json_encode` and `tempnam` never return FALSE (in PeachPie).

Although, `json_encode` should handle incorrect UTF8 byte sequences and resource value types, and return FALSE.
`tempnam()` currently never returns FALSE, and might only throw an exception.

- [ ] json_encode should return false on incorrect UTF8 byte sequences and resource values (unless `JSON_THROW_ON_ERROR` is specified)
- [ ] tempnam() should be reviewed that it always succeeds

**Comment by roberthusak** [comment]
`json_encode` should be fixed now. If `tempnam` returns, the result must be correct. However, there might be a case when it throws a `SecurityException` from https://github.com/peachpiecompiler/peachpie/blob/aee1e2b3c4f9eca492ffbdad7fe0b34187ee337e/src/Peachpie.Library/FileSystem.Path.cs#L289 or an `IOException` from https://github.com/peachpiecompiler/peachpie/blob/aee1e2b3c4f9eca492ffbdad7fe0b34187ee337e/src/Peachpie.Library/FileSystem.Path.cs#L294 I can imagine it happenning in a case when the process doesn't have the access to the temp folder. Should we replace them by returning FALSE instead?

**Comment by jakubmisek** [comment]
thanks @roberthusak!

`tempnam()` should be fixed as well. I can see only one case when it might happen - provided directory is "full" or restricted, and system temp directory is also full (NTFS only).
